onnx/defs/nn/defs.cc

```diff
@@ -2487,4 +2487,194 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  // we need at least three inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 3)) {
+    return;
+  }
+
+  // validating 'input' input
+  auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+  if ((input_shape.dim_size() < 2) || (input_shape.dim_size() > 3)) {
```

Comment:
add similar checks for the other 2 inputs "output_shape" and "kernal_shape"?  to validate the dim_size()==1

Reply:
the check is done on line 2511 by checking `output_shape.size()`